src/utils/imageUtilities.js

```diff
@@ -162,12 +162,36 @@ function imgToPixelArray(img) {
   return Array.from(imgData.data);
 }
 
+/**
+ * Extract common logic from models accepting video input.
+ * Makes sure that the video data has loaded.
+ * Optionally can wait for the next frame every time the function is called.
+ * Will resolve immediately if the input is an image or undefined.
+ * @param {InputImage | undefined} input
+ * @param {boolean} nextFrame
+ * @returns {Promise<void>}
+ */
+async function mediaReady(input, nextFrame) {
+  if (input && isVideo(input)) {
+    if (nextFrame) {
+      await tf.nextFrame();
+    }
+    if (input.readyState === 0) {
+      await new Promise((resolve, reject) => {
+        input.addEventListener('error', () => reject(input.error));
+        input.addEventListener('loadeddata', () => resolve());
+      });
+    }
+  }
+}
+
```

Comment:
since this utility is in the `imageUtilities` it could also be totally fine as is.

Reply:
We will want to add a check to ensure that images are loaded.  It takes that burden off of the user and prevents errors where “pixels provided to tf.browser.fromImage is null”. Similar setup to the video check but with different properties: https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/complete

I assume that there is a similar check that we need to do for microphone and audio inputs but I need to look up the properties.

If we have those three checks then I have to thing about what makes the most sense. Whether there is a single mediaReady function which can be called with all types, or if there are separate ones for audio and image where a video is both.

Since audio and video are so similar (both HTMLMediaElement) then probably one function is better?

Reply:
@joeyklee I added support for HTMLImageElement and HMTLAudioElement into the same function.